core/src/main/scala/kafka/server/DynamicBrokerConfig.scala

```diff
@@ -936,7 +939,10 @@ class DynamicListenerConfig(server: KafkaServer) extends BrokerReconfigurable wi
     if (listenersAdded.nonEmpty)
       server.socketServer.addListeners(listenersAdded)
 
-    server.kafkaController.updateBrokerInfo(server.createBrokerInfo)
+    server match {
+      case kafkaServer: KafkaServer => kafkaServer.kafkaController.updateBrokerInfo(kafkaServer.createBrokerInfo)
+      case _ =>
```

Comment:
Can you log an error message here?  This something we will need to implement (re-registration on listener change) but we don't have time right now.